utils/ioutil.go

```diff
@@ -74,6 +96,55 @@ func (util *impl) Download(URL string, destPath string) error {
 	return out.Sync()
 }
 
+// FetchJSON downloads JSON from URL and returns the bytes in the response.
+func (util *impl) FetchJSON(URL string) ([]byte, error) {
+	resp, err := util.httpClient.Get(URL)
+	if err != nil {
+		return nil, err
+	}
+	defer Close(resp.Body)
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewHTTPError(resp, fmt.Sprintf("Downloading %s failed. Status code %v", URL, resp.StatusCode))
+	}
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+// GetPluginMetas downloads the metadata for each plugin in plugins. If specified,
+// defaultRegistry is used as the registry for plugins that do not specify their registry.
+// If defaultRegistry is empty, and any plugin does not specify a registry, an error is returned.
+func (util *impl) GetPluginMetas(plugins []model.PluginFQN, defaultRegistry string) ([]model.PluginMeta, error) {
+	metas := make([]model.PluginMeta, len(plugins))
+	for idx, plugin := range plugins {
+		log.Printf("Fetching plugin meta.yaml for %s:%s", plugin.ID, plugin.Version)
+		var pluginURL string
+		if plugin.Registry != "" {
+			pluginURL = fmt.Sprintf("%s/%s/%s/meta.yaml", plugin.Registry, plugin.ID, plugin.Version)
+		} else {
+			if defaultRegistry == "" {
+				return nil, errors.New("plugin does not specify registry and no default is provided")
+			}
+			pluginURL = fmt.Sprintf("%s/%s/%s/meta.yaml", defaultRegistry, plugin.ID, plugin.Version)
+		}
+		pluginRaw, err := util.FetchJSON(pluginURL)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to fetch plugin meta.yaml from registry: %s", err)
+		}
+
+		var pluginMeta model.PluginMeta
+		if err := yaml.Unmarshal(pluginRaw, &pluginMeta); err != nil {
+			return nil, fmt.Errorf("Failed to unmarshal downloaded meta.yaml: %s", err)
```

Comment:
Please consider adding plugin id or even plugin URL to find broken meta.yaml easier.

Reply:
Added.